docker/complement/conf/start_for_complement.sh

```diff
@@ -15,26 +19,49 @@ export SYNAPSE_SERVER_NAME=${SERVER_NAME}
 # No need to report stats here
 export SYNAPSE_REPORT_STATS=no
 
-# Set postgres authentication details which will be placed in the homeserver config file
-export POSTGRES_PASSWORD=somesecret
-export POSTGRES_USER=postgres
-export POSTGRES_HOST=localhost
-
-# Specify the workers to test with
-export SYNAPSE_WORKER_TYPES="\
-    event_persister, \
-    event_persister, \
-    background_worker, \
-    frontend_proxy, \
-    event_creator, \
-    user_dir, \
-    media_repository, \
-    federation_inbound, \
-    federation_reader, \
-    federation_sender, \
-    synchrotron, \
-    appservice, \
-    pusher"
+
+case "$SYNAPSE_COMPLEMENT_DATABASE" in
+  postgres)
+    # Set postgres authentication details which will be placed in the homeserver config file
+    export POSTGRES_PASSWORD=somesecret
+    export POSTGRES_USER=postgres
+    export POSTGRES_HOST=localhost
+    export START_POSTGRES=true
```

Comment:
The script can be called multiple times so something as destructive as moving a config file would need something to move it back (I did consider it, but I didn't like it because of this reason). Sure, you could argue that the value of the env vars *probably* won't change between the different runs, but I wasn't keen.